lib/puppet/pops/pcore.rb

```diff
@@ -32,50 +32,6 @@ def self.annotate(instance, annotations_hash)
   end
 
   def self.init_env(loader)
-    if Puppet[:tasks]
-      add_object_type('Task', <<-PUPPET, loader)
-        {
-          attributes => {
-            # Fully qualified name of the task
-            name => { type => Pattern[/\\A[a-z][a-z0-9_]*(?:::[a-z][a-z0-9_]*)*\\z/] },
-
-            # List of implementations with requirements
-            implementations => { type => Array[Struct[name => String, path => String, Optional[requirements] => Array[String]], 1] },
```

Comment:
Add `files` in and next to `implementations`?

Reply:
To preserve the existing structure, we'd probably want `files` to include an extra `path` parameter. We also need to add `input_method` to the `implementations` struct.

Reply:
https://tickets.puppetlabs.com/browse/PUP-5942 will enable open-ended structs later. For now we'll have to drop some validation.